File: server/src/main/kotlin/org/bitlap/server/core/BitlapOperationFactory.kt
package org.bitlap.server.core

import org.bitlap.net.operation.OperationFactory
import org.bitlap.net.operation.OperationType
import org.bitlap.net.operation.operations
import org.bitlap.net.session.Session

class BitlapOperationFactory : OperationFactory {
    override fun create(parentSession: Session, opType: OperationType, hasResultSet: Boolean): operations.Operation {
        return BitlapOperation(parentSession, opType, hasResultSet)
    }
}

File: server/src/main/scala/org/bitlap/server/core/BitlapOperationFactory.scala
/* Copyright (c) 2022 bitlap.org */
package org.bitlap.server.core

import org.bitlap.network.operation.{ operations, OperationFactory }
import org.bitlap.network.operation.OperationType.OperationType
import org.bitlap.network.session.Session

/**
 * @author 梦境迷离
 * @version 1.0,2021/12/3
 */
class BitlapOperationFactory extends OperationFactory {

  override def create(
    parentSession: Session,
    opType: OperationType,
    hasResultSet: Boolean
  ): operations.Operation =
    BitlapOperation(parentSession, opType, hasResultSet)
}
